blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx

```diff
@@ -144,13 +149,31 @@ export class RunDetailsPipeline extends Component {
         const jenkinsEvent = event.jenkins_event;
         // we are using try/catch to throw an early out error
         try {
-            if (event.pipeline_run_id !== this.props.result.id) {
+            const runId = this.props.result.id;
+            // we get events from the pipeline and the job channel, they have different naming for the id
+            if (event.pipeline_run_id !== runId && event.jenkins_object_id !== runId) {
                 // console.log('early out');
                 throw new Error('exit');
             }
             // we turn on refetch so we always fetch a new Node result
             const refetch = true;
+            const refetchNodes = () => {
+                delete this.mergedConfig.node;
+                setTimeout(fetchNodes({ ...this.mergedConfig, refetch }), 50);
```

Comment:
I answered that one a bit above

> I was trying/adopted it from the PR of the rerun button, but it did not really fixed the glitches that I can sometime see

The theory is that given a certain timeout our store would fetch something more reliable. However the bigger problem is that if have a sequence like:

- job_run_queue_enter
- job_run_queue_left
- pipeline_start

which will fire three requests but since they are async we can not be sure that the will return in that order.